src/node_api.cc

```diff
@@ -2736,6 +2736,166 @@ napi_status napi_get_version(napi_env env, uint32_t* result) {
   return napi_clear_last_error(env);
 }
 
+// Conclude a promise either by rejecting it or by resolving it.
+static napi_value napi_conclude_promise(napi_env env,
+                                        napi_callback_info info,
+                                        bool is_resolve) {
+  napi_status status;
+  size_t argc = 1;
+  void *data;
+  napi_value undefined = NULL;
+  napi_value final_value = NULL;
+  v8::Local<v8::Promise::Resolver> v8_resolver;
+
+  // We will return undefined
+  status = napi_get_undefined(env, &undefined);
+  if (status != napi_ok) {
+    goto error;
+  }
+
+  // Retrieve the resolver as well as the value with which we will conclude the
+  // promise.
+  status = napi_get_cb_info(env, info, &argc, &final_value, NULL, &data);
+  if (status != napi_ok) {
+    goto error;
+  }
+
+  // Retrieve the V8 resolver corresponding to this promise.
+  v8_resolver =
+  ((v8::Persistent<v8::Promise::Resolver> *)data)->Get(env->isolate);
+
+  // Call the appropriate method (Resolve/Reject) with the value that will
+  // conclude the promise.
+  if (is_resolve) {
+    v8_resolver->Resolve(env->isolate->GetCurrentContext(),
+    v8impl::V8LocalValueFromJsValue(final_value));
```

Comment:
Can you try to align the arguments vertically? (If that brings the line over 80 chars: wrap after the `(` and indent both arguments with 4 spaces)